### ISSUE TITLE
poscar utilities fail when output and input file are the same

### ISSUE THREAD

**Issue by fkraushofer** [viperleed/viperleed#285]
I just realized that the following fails:
`viperleed poscar sort_by_z --infile POSCAR --outfile POSCAR`
Tested for a valid POSCAR (the utility works if the output goes to a different file); error message is:
```
Invalid unit-cell vectors. Expected shape (3, 3), found (0,) instead
usage: sort_by_z [-h] [--version] [--infile INFILE] [--outfile OUTFILE] [-v]
                 [-r]
sort_by_z: error: Failed to read POSCAR. Stopping. Info: Invalid unit-cell vectors. Expected shape (3, 3), found (0,) instead
```
The POSCAR file is completely empty afterwards.

It seems a safe bet that this is because the outfile is opened (clearing it) before reading (of the same file) is done. The easy fix would be to read in the entire slab, then close the infile, do the processing in memory, and only then open the outfile for writing. I'm not sure if it's possible for io-stream-based calls (`viperleed poscar sort_by_z <POSCAR >POSCAR`), but I feel there should be at least some option that lets you make in-place modifications on a file, without having to copy and rename. At the very least, we should try to catch this in a way that doesn't result in effectively deleting the input file.

**Comment by amimre** [comment]
Alright, that's a clear bug... I'll look into it

**Comment by michele-riva** [comment]
Perhaps an easy solution: open the output file in rw mode rather than r only

**Comment by amimre** [comment]
Fixed in #286.
